databricks/koalas/groupby.py

```diff
@@ -1401,6 +1401,67 @@ def ffill(self, limit=None):
 
     pad = ffill
 
+    def head(self, n=5):
+        """
+        Return first n rows of each group.
+
+        Returns
+        -------
+        DataFrame or Series
+
+        Examples
+        --------
+        >>> df = ks.DataFrame({'a': [1, 1, 1, 1, 2, 2, 2, 3, 3, 3],
+        ...                    'b': [2, 3, 1, 4, 6, 9, 8, 10, 7, 5],
+        ...                    'c': [3, 5, 2, 5, 1, 2, 6, 4, 3, 6]},
+        ...                   columns=['a', 'b', 'c'],
+        ...                   index=[7, 2, 4, 1, 3, 4, 9, 10, 5, 6])
+        >>> df
+            a   b  c
+        7   1   2  3
+        2   1   3  5
+        4   1   1  2
+        1   1   4  5
+        3   2   6  1
+        4   2   9  2
+        9   2   8  6
+        10  3  10  4
+        5   3   7  3
+        6   3   5  6
+
+        >>> df.groupby('a').head(2)
+            a   b  c
+        7   1   2  3
+        2   1   3  5
+        10  3  10  4
+        5   3   7  3
+        3   2   6  1
+        4   2   9  2
+
+        >>> df.groupby('a')['b'].head(2)
+        7      2
+        2      3
+        10    10
+        5      7
+        3      6
+        4      9
+        Name: b, dtype: int64
+        """
+        groupkeys = self._groupkeys
+        tmp_col = '__row_number__'
+        sdf = self._kdf._sdf
+        window = Window.partitionBy([s._scol for s in groupkeys]) \
+                       .orderBy(F.monotonically_increasing_id())
+        sdf = sdf.withColumn(tmp_col, F.row_number().over(window)).filter(F.col(tmp_col) <= n)
```

Comment:
I think we dont need the temp column. You can just use row_number at filter. We should avoid to use the temp column to avoid the name conflict although this a reserved column name in Koalas.

Reply:
i have tried, but we can't use `F.row_number()` directly at filter in this case, since we have to use window function for partition by group which is not supported in filter clause.

```python
>>> sdf.filter(F.row_number().over(window))
Traceback (most recent call last):
pyspark.sql.utils.AnalysisException: 'It is not allowed to use window functions inside WHERE and HAVING clauses;'
```

so i tried like below, but they need also temporal column.
```python
>>> sdf.select("*", F.row_number().over(window).alias('__row_num__')).filter(F.col('__row_num__') < 5)
```

so finally, i think there is only way to avoid using `tmp_col` is using `F.row_number` in `when` clause, after then filter out the null value with `dropna()` which we're doing latest PR.

```python
sdf.select("*", F.when(F.row_number().over(window) <= n, True)).dropna()
```

what are you think?

Reply:
Shall we use the temp column if we can't use `F.row_number()` directly at filter?
It's hard to read.